randovania/layout/description_migration.py

```diff
@@ -442,6 +442,34 @@ def _migrate_v23(data: dict) -> dict:
     return data
 
 
+def _migrate_v24(data: dict) -> dict:
+    game_modifications = data["game_modifications"]
+
+    for game in game_modifications:
+        game_name = game["game"]
+        if game_name != "am2r":
+            continue
+
+        new_ammo_mapping = {
+            "Missile Expansion": "Missile Tank",
+            "Super Missile Expansion": "Super Missile Tank",
+            "Power Bomb Expansion": "Power Bomb Tank",
+        }
+
+        starting_items = game["starting_equipment"]["pickups"]
+        for pickup in starting_items:
```

Comment:
```python
game["starting_equipment"]["pickups"] = [
    new_ammo_mapping.get(pickup, pickup)
    for pickup in game["starting_equipment"]["pickups"]
]
```
I'm really concerned with changing the list while you iterate it. It should have raised an exception I believe, but since it does maybe it's skipping an element?

Alternatively could iterate with `enumerate` and you just change the item inplace.